src/java.desktop/unix/native/libawt_xawt/awt/gtk3_interface.c

```diff
@@ -2406,6 +2406,10 @@ static gint gtk3_get_color_for_state(JNIEnv *env, WidgetType widget_type,
         } else if (widget_type == MENU_BAR && state_type == GTK_STATE_INSENSITIVE
             && color_type == FOREGROUND) {
             widget_type = MENU;
+        } else if ((widget_type == TEXT_FIELD || widget_type == PASSWORD_FIELD
+            || widget_type == SPINNER_TEXT_FIELD || widget_type == FORMATTED_TEXT_FIELD)
+            && state_type == GTK_STATE_INSENSITIVE && color_type == TEXT_FOREGROUND) {
+            widget_type = TEXT_AREA;
         }
```

Comment:
I guess l2403 has same checks which can be clubbed together
```
if (widget_type == TEXT_FIELD || widget_type == PASSWORD_FIELD
            || widget_type == SPINNER_TEXT_FIELD || widget_type == FORMATTED_TEXT_FIELD) {
        if (state_type == GTK_STATE_SELECTED && color_type == TEXT_BACKGROUND} 
            || state_type == GTK_STATE_INSENSITIVE && color_type == TEXT_FOREGROUND) {
            widget_type = TEXT_AREA;
    }
}
```

What do you think?

Reply:
I think it's better to clubbed them, updated as per suggestion.